core/jvm/src/main/scala/scalaz/zio/lockfree/LockFreeQueue.scala

```diff
@@ -0,0 +1,19 @@
+package scalaz.zio.lockfree
+
+trait LockFreeQueue[A] {
+  val capacity: Int
+
+  def offer(a: A): Boolean
+
+  def poll(): Option[A]
```

Comment:
This will create lots of GC churn. We should change the signature to:

```scala
def poll(default: A): A
```

This is still a "safe" interface, although the most common way to use it will be `poll(null)`.

Reply:
Makes sense, will do.